crates/bevy_render/src/render_resource/pipeline_specializer.rs

```diff
@@ -105,55 +111,72 @@ impl<S: SpecializedMeshPipeline> SpecializedMeshPipelines<S> {
         key: S::Key,
         layout: &MeshVertexBufferLayoutRef,
     ) -> Result<CachedRenderPipelineId, SpecializedMeshPipelineError> {
-        match self.mesh_layout_cache.entry((layout.clone(), key.clone())) {
+        return match self.mesh_layout_cache.entry((layout.clone(), key.clone())) {
             Entry::Occupied(entry) => Ok(*entry.into_mut()),
-            Entry::Vacant(entry) => {
-                let descriptor = specialize_pipeline
-                    .specialize(key.clone(), layout)
-                    .map_err(|mut err| {
-                        {
-                            let SpecializedMeshPipelineError::MissingVertexAttribute(err) =
-                                &mut err;
-                            err.pipeline_type = Some(std::any::type_name::<S>());
-                        }
-                        err
-                    })?;
-                // Different MeshVertexBufferLayouts can produce the same final VertexBufferLayout
-                // We want compatible vertex buffer layouts to use the same pipelines, so we must "deduplicate" them
-                let layout_map = match self
-                    .vertex_layout_cache
-                    .raw_entry_mut()
-                    .from_key(&descriptor.vertex.buffers[0])
-                {
-                    RawEntryMut::Occupied(entry) => entry.into_mut(),
-                    RawEntryMut::Vacant(entry) => {
-                        entry
-                            .insert(descriptor.vertex.buffers[0].clone(), Default::default())
-                            .1
+            Entry::Vacant(entry) => specialize_slow(
+                &mut self.vertex_layout_cache,
+                cache,
+                specialize_pipeline,
+                key,
+                layout,
+                entry,
+            ),
+        };
+
+        #[inline(never)]
```

Comment:
Should this be marked with the cold attribute?

Reply:
Good point, I'll replace `#[inline(never)]` with `#[cold]` to let the compiler make the best decision here.